crypto/s2n_openssl_x509.c

```diff
@@ -114,5 +116,20 @@ S2N_RESULT s2n_openssl_x509_get_cert_info(X509 *cert, struct s2n_cert_info *info
     RESULT_GUARD_OSSL(OBJ_find_sigid_algs(info->signature_nid, &info->signature_digest_nid, NULL),
             S2N_ERR_CERT_TYPE_UNSUPPORTED);
 
+    DEFER_CLEANUP(EVP_PKEY *pubkey = X509_get_pubkey(cert), EVP_PKEY_free_pointer);
```

Comment:
I think `X509_get0_pubkey` might be available in openssl 1.0.2, if you think it's worth it to avoid the allocation.

Reply:
It looks like it's not supported by all libcryptos. For example AWS-LC builds fail with this error
```
error: implicit declaration of function 'X509_get0_pubkey'; did you mean 'X509_get_pubkey'? [-Werror=implicit-function-declaration]
  117 |     EVP_PKEY *pubkey = X509_get0_pubkey(cert);
      |                        ^~~~~~~~~~~~~~~~
      |                        X509_get_pubkey
```